pkg/proxy/roundrobin.go

```diff
@@ -210,14 +236,14 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
-		if !exists || !reflect.DeepEqual(existingEndpoints, validEndpoints) {
+		if !exists || !endpointsAreEqual(existingEndpoints, validEndpoints) {
```

Comment:
That is what I was suggesting.  I thought about StringSet, too, but there's no way I can see to deterministically iterate it in order for a given shuffle.  It's a set, not a list.

I think something like `SortSlice([]string) []string` and `CopySlice([]string) []string` would be nice.  Maybe even cleaner to do pkg/util/slice and make it `slice.SortStrings([]string) []string)` and `slice.CopyStrings([]string) []string` (so we can add more later if needed).

Another attack would be to make something like `util.IndexStrings(slice []string) []*string` which would do the sort on pointers to strings, to avoid copying the input.